frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -67,41 +72,94 @@ export class Gallery extends React.Component {
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.SHOW_APP(this.state.config.slug), handler: this.handleShowApp },
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.DELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentDeleted },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.UNDELETED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted }
+    ])
+
     this.lightboxRotation = new LightboxRotation()
+  }
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  // TLM Handlers
+  handleContentCreatedOrModifiedOrUndeleted = data => {
```

Comment:
A picture can be created with another app. Does that answer your question?

Reply:
Not really, it's more a question of naming (i think), because you have a handleContent**Created**OrModifiedOrUndeleted if there is no handle for it, like:

```
{ entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.CREATED, optionalSubType: TLM_ST.FILE, handler: this.handleContentCreatedOrModifiedOrUndeleted }
```

If it is created by another app, it will be handled by it. Right?

Reply:
I rewrote this PR, I think I fixed this.